pkg/controller/serviceexternalip/controller.go

```diff
@@ -313,82 +458,68 @@ func (c *ServiceExternalIPController) syncService(key apimachinerytypes.Namespac
 		c.externalIPAllocator.IPPoolHasIP(currentIPPool, prevIPAllocation.ip) &&
 		currentIPPool == prevIPAllocation.ipPool &&
 		currentExternalIP == prevIPAllocation.ip.String() {
-		return nil
+		// Only the sharable annotation changes, update it.
+		if allowSharedIP != prevIPAllocation.sharable {
+			c.updateIPAllocation(key, prevIPAllocation.ipPool, prevIPAllocation.ip, allowSharedIP)
+		}
```

Comment:
There will not be an error log if the annotation is removed while the IP was already shared?

Reply:
I plan to limit the annotation for IP request stage only. Checking whether an annotation change is valid requires more code, but we can't do anything to prevent it, so perhaps not worth?

Reply:
That makes sense. I think we can start off like this. If it seems that users of this feature keep running into situations that are difficult to troubleshoot without extra logs, we can look into adding more validation / logs.